glib/src/unichar.rs

```diff
@@ -0,0 +1,239 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{
+    translate::{FromGlib, IntoGlib},
+    UnicodeBreakType, UnicodeScript, UnicodeType,
+};
+
+mod sealed {
+    pub trait Sealed {}
+    impl Sealed for char {}
+}
+
+// rustdoc-stripper-ignore-next
+/// The kind of decomposition to perform
+#[derive(Clone, Copy)]
+pub enum DecompositionKind {
+    // rustdoc-stripper-ignore-next
+    /// Compatbility decomposition
+    Compatibility,
+
+    // rustdoc-stripper-ignore-next
+    /// Canonical decomposition
+    Canonical,
+}
+
+impl DecompositionKind {
+    fn into_bool(self) -> bool {
+        use DecompositionKind::*;
+        match self {
+            Compatibility => true,
+            Canonical => false,
+        }
+    }
+}
```

Comment:
Why not use the From trait here?

Reply:
I can't really recall my thought process here, I'll go with whatever you prefer

Reply:
What is the use case for this?

Reply:
I ended up just removing and inlining the contents of this function in the one place where it is used (`fully_decompose`)